getnacelle/nacelle-js
# Initial Lerna setup

### WHY are these changes introduced?

Addresses [DD-994](https://nacelle.atlassian.net/browse/DD-994)

> OSS Repo Setup: Lerna

### What this does

- initializes [Lerna](https://lerna.js.org/) in [independent mode](https://github.com/lerna/lerna#independent-mode)
- documents Lerna workflows for adding new packages and example projects

### What this doesn't do

This PR does not make an attempt to configure Lerna for CI/CD workflows.

Before that can be done, we need to perform a research spike to understand the requirements and implications around tools like [`semantic-release`](https://www.npmjs.com/package/semantic-release) and practices such as [Conventional Commits](https://www.conventionalcommits.org).

### How to Test

- check out the `DD-994-oss-repo-setup-lerna` branch
- `npm run bootstrap` should result in: 
```
lerna notice cli v4.0.0
lerna info versioning independent
lerna info Bootstrapping 0 package
lerna info Installing external dependencies
lerna info hoist Finished bootstrapping root
lerna info Symlinking packages and binaries
lerna success Bootstrapped 0 package
```
- `npm run create -- some-package-name packages` should scaffold `packages/some-package-name`

### Testing checklist
